### PR TITLE
Add psock device

### PR DESCRIPTION
**Pull request by dberliner**
Adds a new device type for our psock driver, no options are currently passed.

### CODE REVIEW

File: src/function/psock.c [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+}
+
+static int psock_libconfig_export(struct usbg_function *f,
+				  config_setting_t *root)
+{
+	return USBG_SUCCESS;
+}
+
+#define PSOCK_FUNCTION_OPTS			\
+	.alloc_inst = psock_alloc_inst,	\
```
Comment (drbild) [inline src/function/psock.c]: Misaligned terminating `\`

**Review by drbild — APPROVED** [review]
Just the one formatting issue. Otherwise LGTM.

**Comment by drbild** [comment]
@dberliner Rather than merging to `master`, let's create a `xaptum` branch (or `xaprc`) for our fork. Then master can continue to track upstream.

File: src/function/psock.c (hunk 1 above)
Comment (dberliner) [reply src/function/psock.c]: Fixed

**Pull request merged**